Add image for clients with missing or unknown platform

stintel opened the pull request:
When running the UI in dev mode, having a client with missing or unknown platform causes an error. Add a fallback image to fix this.

stintel committed:
Add image for clients with missing or unknown platform

When running the UI in dev mode, having a client with missing or unknown
platform causes an error. Add a fallback image to fix this.

ciaranj reviewed (approved):
Initially this looked out to me, although src=`{ClientImages[client.platform] ?? UnknownImage}` and `alt={client.platform || "Unknown device"}` are *subtly* different. An invalid (from the perspective of the ClientImages constant) client.platform value will return the 'UnknownImage' image, but display the value of the client.platform variable. Only an undefined/null client.platform will display the 'Unknown Device' label. Was this your intent @stintel  ?

edit: Which on further analysis checked the type of client.platform, looks non-nullable, so I don't think it's ever possible for 'unknown Device' to be shown.

stintel commented:
My intent was to fix the error. Probably didn't think hard enough about the implication of the alt change. Have any suggestions to make/keep things consistent?

ciaranj commented:
> My intent was to fix the error. Probably didn't think hard enough about the implication of the alt change. Have any suggestions to make/keep things consistent?

I guess something like: `ClientImages[client.platform] ? client.platform : "Unknown device"` would give you a consistent logic.

stintel committed:
Add image for clients with missing or unknown platform

When running the UI in dev mode, a client with a missing or unknown
platform causes Next.js to reject the undefined image source.

Add a fallback image, normalize missing, null, and empty platform values
to "Unknown", and reflect the optional value in the Client type.
